src/main/java/backtrace/io/Backtrace.java

```diff
@@ -32,21 +32,23 @@
      * Handles the queue of incoming error reports
      */
     void handleBacktraceMessage() {
+        if (queue.isEmpty()) {
```

Comment:
How often do we call this function to check if the queue is empty, or not? The reason why I'm asking is that I'm worry that the background thread will check queue size for every thread tick.

Reply:
it will be called probably once per message because I've added additional lock which should fix this problem with checking queue size on every thread tick